src/DistributedComputations/distributed_fft_tridiagonal_solver.jl

```diff
@@ -0,0 +1,333 @@
+2. Transform along ``z``.
+3. Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
+4. Transform along ``y``.
+5. Transpose + communicate to `storage.xfield` partitioned into `(Rx, Ry)` processes in ``(y, z)``.
+6. Solve the tri-diagonal linear system in the ``x`` direction.
+
+Steps 5 -> 1 are reversed to obtain the result in physical space stored in `storage.zfield` 
+partitioned over ``(x, y)``.
+
+Y - stretched algorithm
+========================
+
+1. `storage.zfield`, partitioned over ``(x, y)`` is initialized with the `rhs`.
+2. Transform along ``z``.
+3. Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
+4. Transpose + communicate to `storage.xfield` partitioned into `(Rx, Ry)` processes in ``(y, z)``.
+5. Transform along ``x``.
+6. Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
+7. Solve the tri-diagonal linear system in the ``y`` direction.
+
+Steps 6 -> 1 are reversed to obtain the result in physical space stored in `storage.zfield` 
+partitioned over ``(x, y)``.
+
+Z - stretched algorithm
+========================
+
+1. `storage.zfield`, partitioned over ``(x, y)`` is initialized with the `rhs`.
+2. Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
+3. Transform along ``y``.
+4. Transpose + communicate to `storage.xfield` partitioned into `(Rx, Ry)` processes in ``(y, z)``.
+5. Transform along ``x``.
+6. Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
+7. Transpose + communicate to `storage.zfield` partitioned into `(Rx, Ry)` processes in ``(x, y)``.
+8. Solve the tri-diagonal linear system in the ``z`` direction.
+
+Steps 7 -> 1 are reversed to obtain the result in physical space stored in `storage.zfield` 
+partitioned over ``(x, y)``.
+
+Algorithm for slab decompositions
+============================================
+
+The 'slab' decomposition works in the same manner while skipping the transposes that
+are not required. For example if the domain is decomposed in ``x``, step 4. and 6. in the above algorithm
+are skipped (and the associated reversed step in the backward transform)
+
+Restrictions
+============
+
+1. Pencil decompositions:
+    - `Ny ≥ Rx` and `Ny % Rx = 0`
+    - `Nz ≥ Ry` and `Nz % Ry = 0`
+    - If the ``z`` direction is `Periodic`, also the ``y`` and the ``x`` directions must be `Periodic`
+    - If the ``y`` direction is `Periodic`, also the ``x`` direction must be `Periodic`
+
+2. Slab decomposition:
+    - same as for two-dimensional decompositions with `Rx` (or `Ry`) equal to one
+
+"""
+function DistributedFourierTridiagonalPoissonSolver(global_grid, local_grid, planner_flag=FFTW.PATIENT)
+    
+    validate_poisson_solver_distributed_grid(global_grid)
+    validate_poisson_solver_configuration(global_grid, local_grid)
+ 
+    irreg_dim = stretched_dimensions(local_grid)[1]
```

Comment:
err... irreg? stretched?

```suggestion
    stretched_dim = stretched_dimensions(local_grid)[1]
```